WordPress/Sniffs/Utils/I18nTextDomainFixerSniff.php

```diff
@@ -0,0 +1,725 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPress\Sniffs\Utils;
+
+use WordPress\AbstractFunctionParameterSniff;
+use PHP_CodeSniffer_Tokens as Tokens;
+
+/**
+ * Comprehensive I18n text domain fixer tool.
+ *
+ * This sniff can:
+ * - Add missing `text-domain`s.
+ * - Replace `text-domain`s based on an array of `old` values to a `new` value.
+ *
+ * Note: Without a user-defined configuration in a custom ruleset, this sniff will be ignored.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   1.2.0
+ */
+class I18nTextDomainFixerSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * A list of tokenizers this sniff supports.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array(
+		'PHP',
+		'CSS',
+	);
+
+	/**
+	 * Old text domain(s) to replace.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @var string[]|string
+	 */
+	public $old_text_domain;
+
+	/**
+	 * New text domain.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @var string
+	 */
+	public $new_text_domain = '';
+
+	/**
+	 * The group name for this group of functions.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @var string
+	 */
+	protected $group_name = 'i18nfixer';
+
+	/**
+	 * The WP Internationalization related functions to target for the replacements.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @var array <string function name> => <int parameter position>
+	 */
+	protected $target_functions = array(
+		'load_textdomain'                        => 1,
+		'load_plugin_textdomain'                 => 1,
+		'load_muplugin_textdomain'               => 1,
+		'load_theme_textdomain'                  => 1,
+		'load_child_theme_textdomain'            => 1,
+		'unload_textdomain'                      => 1,
+
+		'__'                                     => 2,
+		'_e'                                     => 2,
+		'_x'                                     => 3,
+		'_ex'                                    => 3,
+		'_n'                                     => 4,
+		'_nx'                                    => 5,
+		'_n_noop'                                => 3,
+		'_nx_noop'                               => 4,
+		'translate_nooped_plural'                => 3,
+		'_c'                                     => 2, // Deprecated.
```

Comment:
[`ngettext`](https://developer.wordpress.org/reference/hooks/ngettext/), [`_nc`](https://developer.wordpress.org/reference/functions/_nc/) (deprecated), [`__ngettext`](https://developer.wordpress.org/reference/functions/__ngettext/) (deprecated), and maybe others, should be included?

Reply:
> ngettext, _nc (deprecated), __ngettext (deprecated), and maybe others, should be included?

Good call on the last two. Updating the PR as we speak.
Thralling through the `wp-includes/deprecated.php` file which contained these, yielded me two more: [`translate_with_context()`](https://developer.wordpress.org/reference/functions/translate_with_context/) and [`__ngettext_noop`](https://developer.wordpress.org/reference/functions/__ngettext_noop/).

If anyone can find yet more, please comment.

As for the first - that is filter, not a function and I would not expect a plugin/theme to contain a call to such a specific WP core native filter, so I'm not sure we should account for that.
What do you think ?

Reply:
> As for the first - that is filter

Yep, good catch. Ignore that part of my suggestion :-)